olp-cpp-sdk-core/include/olp/core/client/OlpClient.h

```diff
@@ -119,6 +121,59 @@ class CORE_API OlpClient {
                        std::string content_type,
                        CancellationContext context) const;
 
+  /**
+   * @brief Execute the REST request through the network stack
+   * @param path Path that is appended to the base url.
+   * @param method The request HTTP method.
+   * @param query_params Params that is appended to the path url.
+   * @param header_params Headers to customize request.
+   * @param form_params For a POST request, form_params or post_body should be
+   * populated, but not both.
+   * @param post_body For a POST request, form_params or post_body should be
+   * populated, but not both.
+   * This data must not be modified until the request is completed.
+   * @param content_type Content type for the post_body or form_params.
+   * @param callback a function callback to receive the HttpResponse.
+   *
+   * @return A method to call to cancel the request.
+   */
+  CancellationToken CallApi(
+      const std::string& path, olp::http::NetworkRequest::HttpVerb method,
+      const std::multimap<std::string, std::string>& query_params,
+      const std::multimap<std::string, std::string>& header_params,
+      const std::multimap<std::string, std::string>& form_params,
+      const std::shared_ptr<std::vector<unsigned char>>& post_body,
```

Comment:
See if you can use `http::NetworkRequest::RequestBodyType` instead of this.
Same for above method.

Also, shared_ptrs should be passed by copy and moved internally.

Reply:
I'll make it in a follow-up commit